### DISCUSSION TITLE
Help window tooltip **bold** rendering

### DISCUSSION THREAD

**Discussion by Malachov** [microsoft/pylance-release#2921] [Enhancement]
Hello. First thanks for great job about help tooltip. Links, bullets , headings works just great.

Is it possible to add rendering of bold text with asterisks syntax (both in rst as well as in markdown).

Printscreen how its rendered now...

![image](https://user-images.githubusercontent.com/39381691/150185916-d08808f9-45bc-4511-99fc-7d643afdafcb.png)

Note:
I'm not sure whether the content is generated in pylance, if not, just close the issue.

Regards    Daniel

**Comment by judej** [comment]
Thanks for the report. Will investigate this.

**Comment by heejaechang** [comment]
@Malachov can you provide more details on what you want? examples and more explanation will let us understand what you are asking better.

thank you.

**Comment by Malachov** [comment]
Hello, when I understand, that it's very hard to provide correct formatting of python docstrings as there are more ways to do it (rst, google, numpy styles). Still I don't know how the formatting works and how to write docstrings so it can be displayed user friendly. Python default rst doesn't render some things. I found, that most of markdown is formatted correctly. There are also a tools for generating e.g. sphinx documentation even from markdown docstrings. Still there are some basic syntax pieces like bold text in markdown that are not formatted (mainly bold text). If I found for what type of docstrings supported in VS Code i found only some stackoverflow discussions which leads into compromise - sphinx or VS Code tooltip. For me markdown is OK for now. 

So to describe example: On the picture above, instead \*\*bold\*\* I would expect **bold** in VS code help tooltip. Or describe somewhere what language is used for formatting or describe what will be supported in future, so users will not must to change the code in the future.

**Comment by bschnurr** [comment]
checkout numpy's doc strings.

we are expecting google, or reST 

see  https://en.wikipedia.org/wiki/ReStructuredText

we dont expect markdown. but try to make plain text look similar in markdown 
https://github.com/microsoft/pyright/blob/7bb059ecbab5c0c446d4dcf5376fc5ce8bd8cd26/packages/pyright-internal/src/analyzer/docStringUtils.ts#L69

**Reply by BrutalSimplicity** [reply]
In the Wikipedia article linked, it shows proper bold/italic formatting. I think this issue was originally about the inconsistency in the rendering of the docstring format. As @Malachov pointed out. The rendered docstring in vscode doesn't render **bold** or *italic* markings properly. It would seem that since code sections, headers, and lists are rendered properly that bold or italic references would also be included in that group since these are the same in both Markdown and ReST.

**Comment by bschnurr** [comment]
as an enhancement we could support markdown

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by Malachov** [comment]
I remember times when google / numpy / just rst parsing was displayed in a strange way. As I tested on new version, with no configuration what docs style I'm using, everything seems very well now :) .

![image](https://user-images.githubusercontent.com/39381691/173671260-c80d8ffe-307f-4160-8a60-4be7d1456b16.png)

Mostly important, example seems great. RST directives (e.g. images) are ignored, but mostly it's not necessary... sometimes I use also ..code-block if some shell code, but it's also not necessary.
 
 Sorry for unnecessary issue - I checked it on bold as **bold** syntax [as it's documented](https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#inline-markup-and-special-characters-e-g-bold-italic-verbatim) as RST syntax compatible and it came first to my mind. I though that google docstring is a subset and it will work here, so google is not parsed... 

I think current state is OK for me as args and examples are pretty clear. Bold I can substitute with ``

Thanks Dan
